librz/core/cmd/cmd_analysis.c

```diff
@@ -8812,3 +8443,220 @@ RZ_IPI RzCmdStatus rz_analysis_class_vtable_lookup_handler(RzCore *core, int arg
 	ls_free(classes);
 	return RZ_CMD_STATUS_OK;
 }
+
+static RzList *get_xrefs(RzAnalysisBlock *block) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+	RzList *list = NULL;
+	size_t i;
+	for (i = 0; i < block->ninstr; i++) {
+		ut64 ia = block->addr + block->op_pos[i];
+		RzList *xrefs = rz_analysis_xrefs_get_to(block->analysis, ia);
+		rz_list_foreach (xrefs, iter, xref) {
+			if (!list) {
+				list = rz_list_newf(free);
+			}
+			rz_list_push(list, ut64_new(xref->from));
+		}
+	}
+	return list;
+}
+
+static RzList *get_calls(RzAnalysisBlock *block) {
+	RzList *list = NULL;
+	RzAnalysisOp op;
+	ut8 *data = malloc(block->size);
+	if (data) {
+		block->analysis->iob.read_at(block->analysis->iob.io, block->addr, data, block->size);
+		size_t i;
+		for (i = 0; i < block->size; i++) {
+			int ret = rz_analysis_op(block->analysis, &op, block->addr + i, data + i, block->size - i, RZ_ANALYSIS_OP_MASK_HINT);
+			if (ret < 1) {
+				continue;
+			}
+			if (op.type == RZ_ANALYSIS_OP_TYPE_CALL) {
+				if (!list) {
+					list = rz_list_newf(free);
+				}
+				rz_list_push(list, ut64_new(op.jump));
+			}
+			rz_analysis_op_fini(&op);
+			if (op.size > 0) {
+				i += op.size - 1;
+			}
+		}
+	}
+	return list;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_basic_block_info_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	RzAnalysisBlock *bb = rz_analysis_find_most_relevant_block_in(core->analysis, core->offset);
+	if (!bb) {
+		RZ_LOG_ERROR("No basic block at 0x%" PFMT64x "\n", core->offset);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_core_analysis_bb_info_print(core, bb, core->offset, state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_basic_block_list_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	PJ *pj = state->mode == RZ_OUTPUT_MODE_JSON ? state->d.pj : NULL;
+	RzTable *table = state->mode == RZ_OUTPUT_MODE_TABLE ? state->d.t : NULL;
+
+	if (pj) {
```

Comment:
Use rz_cmd_state_output_start_array and also set_columnsf (see other code for how to use)